### PR TITLE
feat(@juxio/react-primitives): added headless select component

### PR DESCRIPTION
**Pull request by tnipri**

### CODE REVIEW

File: packages/primitives/react/src/primitives/select/Select.tsx [hunk 1]
```diff
@@ -0,0 +1,672 @@
...
+}
+
+type Direction = 'ltr' | 'rtl';
+
+/****************
+ * UTILS
+ ****************/
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function isValueEmpty(value?: any | any[]) {
```
Comment (hamatoyogi) [inline packages/primitives/react/src/primitives/select/Select.tsx:53]: would be good to have unit tests for helper functions

File: packages/primitives/react/src/primitives/select/Select.tsx [hunk 2]
```diff
@@ -0,0 +1,672 @@
...
+    placement: placement,
+    whileElementsMounted: autoUpdate,
+    onOpenChange: setOpen,
+    middleware: [
+      offset({ mainAxis: sideOffset, alignmentAxis: alignOffset }),
+      flip(),
+      size({
+        apply({ rects, elements }) {
+          elements.floating.style.setProperty(
+            '--jux-select-trigger-width',
```
Comment (hamatoyogi) [inline packages/primitives/react/src/primitives/select/Select.tsx:166]: might be worth putting the string value in a constant

File: packages/primitives/react/src/primitives/select/Select.tsx [hunk 3]
```diff
@@ -0,0 +1,672 @@
...
+
+const OPTIONS_NAME = 'Jux.Select.Options';
+
+export type SelectOptionsProps = React.ComponentPropsWithoutRef<
+  typeof BasePrimitive.div
+>;
+
+type SelectOptionsElement = React.ElementRef<typeof BasePrimitive.div>;
+
+const Options = React.forwardRef<SelectOptionsElement, SelectOptionsProps>(
```
Comment (hamatoyogi) [inline packages/primitives/react/src/primitives/select/Select.tsx:397]: just a reminder to us that React 19 we will [no longer need to `forwardRef`](https://react.dev/blog/2024/04/25/react-19#ref-as-a-prop)

File: packages/primitives/react/src/primitives/select/select.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,318 @@
+import React, { useState } from 'react';
+import type { Meta, StoryObj } from '@storybook/react';
+import * as Select from './Select';
+import { css } from '@juxio/react-styled';
```
Comment (hamatoyogi) [inline packages/primitives/react/src/primitives/select/select.stories.tsx:4]: we should add component tests and a11y tests for the component stories:
https://storybook.js.org/docs/writing-tests/component-testing#write-a-component-test

**Review by hamatoyogi — APPROVED** [review]
nice work on this! 

see comments please.

File: packages/css/src/base.ts [hunk 5]
```diff
@@ -181,29 +181,28 @@ export type CSSPropertiesWithCustomValues<
   Utilities extends BaseProps = EmptyObject,
   Conditions extends BaseProps = EmptyObject,
 > = {
-  [K in keyof CSSProperties]: BaseStyles[K];
+  [K in Prefix<'&', CSS.Pseudos>]?: CSSPropertiesWithCustomValues<
```
[the remark this answers is not in the document]
Comment (tnipri) [reply packages/css/src/base.ts]: Yes but your syntax is incorrect, it should be

`{
         '& p': {
            color: 'red',
         },
    }`

File: packages/primitives/react/src/primitives/select/Select.tsx (hunk 2 above)
Comment (tnipri) [reply packages/primitives/react/src/primitives/select/Select.tsx]: Added constant

File: packages/primitives/react/src/primitives/select/Select.tsx (hunk 3 above)
Comment (tnipri) [reply packages/primitives/react/src/primitives/select/Select.tsx:399]: Thanks. For backward compatibility, let's keep it as is and modify it once React 19 is adopted @hamatoyogi

File: packages/primitives/react/src/primitives/select/select.stories.tsx (hunk 4 above)
Comment (tnipri) [reply packages/primitives/react/src/primitives/select/select.stories.tsx:4]: Definitely, on our todo-list

File: packages/primitives/react/src/primitives/select/Select.tsx (hunk 1 above)
Comment (tnipri) [reply packages/primitives/react/src/primitives/select/Select.tsx]: Same

**Pull request merged**